assets/js/modules/analytics-4/utils/data-mock.js

```diff
@@ -862,46 +905,57 @@ export function getAnalytics4MockPivotResponse( options ) {
 	args.pivots.forEach( ( { fieldNames, limit } ) => {
 		// We only support one dimension for each pivot in our current pivot report implementation
 		// so we can choose the first fieldNames value as our dimension.
-		const dimension = fieldNames[ 0 ];
 
-		if (
-			dimension &&
-			typeof ANALYTICS_4_DIMENSION_OPTIONS[ dimension ] === 'function'
-		) {
-			// Generates a stream (an array) of dimension values using a function associated with the current dimension.
-			streams.push(
-				new Observable( ( observer ) => {
-					// The limit here is the pivot limit as limits are set within the pivot level, never at the root of the report.
-					for ( let i = 1; i <= limit; i++ ) {
-						const dimensionValue =
-							ANALYTICS_4_DIMENSION_OPTIONS[ dimension ]( i );
-						if ( dimensionValue ) {
-							observer.next( dimensionValue );
-						} else {
-							break;
+		fieldNames.forEach( ( dimension ) => {
+			// If an array of filter values is provided for the dimension, use that to generate the stream.
+			if (
+				dimension &&
+				args?.dimensionFilters?.[ dimension ]?.some(
```

Comment:
Hey @ankitrox I just realised `every()` would be more appropriate here:

```suggestion
				args?.dimensionFilters?.[ dimension ]?.every(
```